command_line_assistant/utils/files.py

```diff
@@ -93,3 +100,94 @@ def write_file(contents: Union[str, bytes], path: Path, mode: int = 0o600) -> No
             path,
             str(e),
         )
+
+
+class NamedFileLock:
+    """File-based lock mechanism for general locking purposes.
+
+    This class provides a way to track state across processes
+    using a lock file. The lock file contains the PID of the capturing process.
+
+    Usage:
+        with NamedFileLock() as lock:
+            # Execute your operations
+        # Lock is automatically released
+    """
+
+    def __init__(self, name: str = "lock"):
+        """Initialize the named file lock mechanism."""
+        self._pid = os.getpid()
+        self._name = name
+        self._lock_file = Path(get_xdg_state_path(), name + ".pid")
```

Comment:
I feel like the suffix should more clearly identify the purpose of the file. Writing the PID to the file makes sense, but we should name the file `something.lock`. If I were trying to clean up files and found a file called `.lock`, it would tell me more clearly that the file is used for some sort of locking mechanism rather than a `.pid` file.

Reply:
Ah! That's a good point. Let me update it

Reply:
```suggestion
        self._lock_file = Path(get_xdg_state_path(), name + ".lock")
```